clients/gitlabrepo/webhook_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitlabrepo
+
+import (
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/xanzy/go-gitlab"
+
+	"github.com/ossf/scorecard/v4/clients"
+)
+
+type stubTripper struct {
+	responsePath string
+}
+
+func (s stubTripper) RoundTrip(_ *http.Request) (*http.Response, error) {
+	f, err := os.Open(s.responsePath)
+	if err != nil {
+		//nolint:wrapcheck
+		return nil, err
+	}
+	return &http.Response{
+		Status:     "200 OK",
+		StatusCode: http.StatusOK,
+		Body:       f,
+	}, nil
+}
+
+func Test_listWebhooks(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name         string
+		responsePath string
+		want         []clients.Webhook
+		wantErr      bool
+	}{
+		{
+			name:         "valid webhook",
+			responsePath: "./testdata/valid-webhook",
+			want: []clients.Webhook{
+				{
+					ID:             1,
+					Path:           "http://example.com/hook",
+					UsesAuthSecret: true,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:         "invalid webhook",
```

Comment:
it's not so much the webhook is invalid, but rather that `listWebhooks` returns an error. Should the name match that?

What do you think about: "failure fetching webhooks"

Reply:
👍